source/shared/core_sqlsrv.h

```diff
@@ -2040,7 +2040,21 @@ namespace core {
         SQLRETURN r;
         SQLSMALLINT num_cols;
         r = ::SQLNumResultCols( stmt->handle(), &num_cols );
-        
+
+        // Workaround for a bug in unixODBC: after SQLExecDirect returns SQL_NO_DATA,
+        // r = ::SQLNumResultCols( stmt->handle(), &num_cols );  
+        // returns r=-1 (SQL_ERROR) and error HY010 (Function sequence error)
+        // but it should have succeeded with r=0 (SQL_SUCCESS) and no error
+        // instead of throwing an exception, return 0 if the r=-1, stament has been executed, and has a HY010 error
+        // (HY010 error should not return if stmt->execute is true)
+#ifndef _WIN32
+        sqlsrv_error_auto_ptr error;
+        if ( stmt->current_results != NULL ) {
```

Comment:
You don't have to make changes, but you can actually declare the variable error inside the IF block ;)